codex/utils.nim

```diff
@@ -13,3 +13,25 @@ func roundUp*[T](a, b : T): T =
   ## Round up 'a' to the next value divisible by 'b'
   divUp(a,b) * b
 
+template newSeqWith*(len: uint64, init: untyped): untyped =
```

Comment:
The difference is that `sequtils.newSeqWith` does not allow a length of `uint64`. I was not aware that the max array size was `high(int)` which makes sense why `newSeqWith` did not allow for a `uint64` length.

We need to be able to randomly shuffle a number of slots. The issue, however, is that the number of slots is a `uint64`. It seems like we have a couple of options:
1. Truncate slots to an int, and use `sequtils.newSeqWith` to shuffle. This will reduce the capacity of slots in a contract in the client code which is currently defined in the contracts as a `uint256`.
2. Split up all `uint64` slots into a number of `seqs` (with len of `high(int)`), then shuffle each of those seqs individually, and concatenate them at the end.

I'd honestly like to go with option 1, and use `int` for `StorageAsk.slots` as this will make things quite a lot simpler and prevent any under/overflow errors with `uints`, however the main concern is whether or not `high(int)` (= `9223372036854775807` on my machine) slots is enough 😆. If we do decide to use `int`, then we'd need to change the type in the contracts as well.

Reply:
Yeah, we should probably tighten all those uint256's in the contract. For number of slots `uint16` should be more than enough, maybe even `uint8`?

Reply:
And on the Nim side, use `int` instead of `uint`?

Reply:
I would use the same type as in the contract, but convert to `int` when needed (for instance when using with `newSeqWith`).